File: src/main/java/com/star/easydoc/service/variable/impl/AuthorVariableGenerator.java
package com.star.easydoc.service.variable.impl;

import com.intellij.openapi.components.ServiceManager;
import com.intellij.psi.PsiElement;
import com.star.easydoc.config.EasyJavadocConfigComponent;
import com.star.easydoc.model.EasyJavadocConfiguration;
import com.star.easydoc.service.variable.VariableGenerator;

/**
 * @author <a href="mailto:wangchao.star@gmail.com">wangchao</a>
 * @version 1.0.0
 * @since 2019-12-07 23:16:00
 */
public class AuthorVariableGenerator implements VariableGenerator {
    private EasyJavadocConfiguration config = ServiceManager.getService(EasyJavadocConfigComponent.class).getState();

    @Override
    public String generate(PsiElement element) {
        return config.getAuthor();
    }
}

File: easydoc-kdoc/src/main/kotlin/com/star/easydoc/kdoc/service/variable/impl/AuthorVariableGenerator.kt
package com.star.easydoc.kdoc.service.variable.impl

import com.intellij.psi.PsiElement

/**
 * @author [wangchao](mailto:wangchao.star@gmail.com)
 * @version 1.0.0
 * @since 2019-12-07 23:16:00
 */
class AuthorVariableGenerator : AbstractVariableGenerator() {
    override fun generate(element: PsiElement): String {
        return config.author
    }
}